custom_components/samsungtv_smart/config_flow.py

```diff
@@ -484,6 +486,18 @@ def _async_option_form(self):
                 CONF_SYNC_TURN_ON,
                 description={"suggested_value": options.get(CONF_SYNC_TURN_ON, [])},
             ): EntitySelector(select_entities),
+            vol.Required(
+                CONF_POWER_ON_METHOD,
+                default=options.get(
+                    CONF_POWER_ON_METHOD, str(PowerOnMethod.WOL.value)
+                ),
+            ): SelectSelector(_dict_to_select(POWER_ON_METHODS)),
+            vol.Required(
+                CONF_POWER_ON_SERVICE_DATA,
+                default=options.get(
+                    CONF_POWER_ON_SERVICE_DATA, "{}"
+                ),
+            ): cv.string,
```

Comment:
In this way we are going to provide a `SmartThings` option also if `SmartThings` is not enabled (not good) and we also add a text field that should be useless in some cases. This will probably confuse "normal" user and cause false issues, because I'm trying to simplify option flow (I will introduce some changes shortly), I do not like very match this approach.

I suggest to add only the text field in the advanced option:

- if it contain some values, the startup based on service will be used (may be you could also hide the power on method field in main form in this case)
- if empty, everything will normally